LambdaEngine/Include/ECS/ComponentArray.h

```diff
@@ -134,6 +141,59 @@ namespace LambdaEngine
 		m_EntityToIndex.erase(indexItr);
 	}
 
+	template<typename Comp>
+	inline uint32 ComponentArray<Comp>::SerializeComponent(const Comp& component, uint8* pBuffer, uint32 bufferSize) const
+	{
+		/*	ComponentSerializationHeader is written to the beginning of the buffer. This is done last, when the size of
+			the serialization is known. */
+		struct ComponentSerializationHeader
+		{
+			uint32 TotalSerializationSize;
+			uint32 TypeHash;
+		};
```

Comment:
To make sure no padding is added by the compiler in the future if adding more data to the header

```suggestion
        #pragma pack(push, 1)
		struct ComponentSerializationHeader
		{
			uint32 TotalSerializationSize;
			uint32 TypeHash;
		};
        #pragma pack(pop)
```